### PR TITLE
docs: adding future plans

### PR DESCRIPTION
**Pull request by holtgrewe**

### CODE REVIEW

File: src/doc_futureplans.rst [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+--------------
+
+There is some technical debt, some notable items.
+
+Automated Tests
+    We need more automated tests in various areas throughout the codebase.
+
+Python Type Annotations
+    We should modernize the codebase by comprehensive use of Python type annotations.
+    Notable, this will require type annotations for sodar-core.
```
Comment (xiamaz) [inline src/doc_futureplans.rst:24]: Do we already have mypy in our CI? Otherwise maybe mention that.

File: src/doc_futureplans.rst [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+
+Backing Server Protobuf Migration
```
Comment (xiamaz) [inline src/doc_futureplans.rst:26]: Backend

**Review by xiamaz — APPROVED** [review]
LGTM

File: src/doc_futureplans.rst (hunk 2 above)
Comment (holtgrewe) [reply src/doc_futureplans.rst:26]: The back backend services are introduced as backing in the architecture.

File: src/doc_futureplans.rst (hunk 1 above)
Comment (holtgrewe) [reply src/doc_futureplans.rst:24]: IMO that goes without saying. Actually, pyright is the cool new engine to use ;-)

**Pull request merged**